src/Hid.Shared/Hid+SafePreparsedDataHandle.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using Microsoft.Win32.SafeHandles;
+
+    /// <content>
+    /// Contains the <see cref="SafePreparsedDataHandle"/> nested class.
+    /// </content>
+    public static partial class Hid
+    {
+        public class SafePreparsedDataHandle : SafeHandleMinusOneIsInvalid
+        {
+            public SafePreparsedDataHandle()
+                : base(true)
+            {
+            }
+
+            public SafePreparsedDataHandle(IntPtr preexistingHandle, bool ownsHandle)
+                : base(ownsHandle)
+            {
+                this.handle = preexistingHandle;
```

Comment:
tip: *maybe* `SetHandle` is better to call here rather than setting the field directly. At the moment it appears they're equivalent, and maybe they'll always be. But since .NET itself sets it via the method, I suggest we do the same.